ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/API.hs

```diff
@@ -151,7 +148,10 @@ data ImmutableDB m = ImmutableDB
       :: HasCallStack
       => EpochSlot
       -> EpochSlot
-      -> m (Either ImmutableDBError (Iterator m))
+      -> m (Iterator m)
+
+    -- Throw immutable DB errors
+  , immutableDbErr :: ErrorHandling ImmutableDBError m
```

Comment:
Suggestion:
```haskell
    -- | How to throw an 'ImmutableDBError'.
  , hasImmutableDBErr :: ErrorHandling ImmutableDBError m
```

Also, I have noticed we don't actually use this right now. But I guess I might need it in my q-s-m tests.